python/benchmark/benchmark/bench_dbscan.py

```diff
@@ -0,0 +1,304 @@
+    def _supported_class_params(self) -> Dict[str, Any]:
+        from pyspark.ml.clustering import KMeans
+
+        params = inspect_default_params_from_func(
+            KMeans.__init__,
+            [
+                "distanceMeasure",
+                "featuresCol",
+                "labelCol",
+                "predictionCol",
+                "probabilityCol",
+                "rawPredictionCol",
+                "weightCol",
+                "leafCol",
+            ],
+        )
+        params["seed"] = 1
+        params["eps"] = float
+        params["min_samples"] = int
+        return params
+
+    def _parse_arguments(self, argv: List[Any]) -> None:
+        """Override to set class params based on cpu or gpu run (umap or pca)"""
+        pp = pprint.PrettyPrinter()
+
+        self._args = self._parser.parse_args(argv)
+        print("command line arguments:")
+        pp.pprint(vars(self._args))
+
+        if self._args.num_cpus > 0:
+            supported_class_params = self._supported_class_params()
+            supported_class_params.pop("eps", None)
+            supported_class_params.pop("min_samples", None)
+        else:
+            supported_class_params = {
+                "eps": float,
+                "min_samples": int,
+            }
+        self._class_params = {
+            k: v
+            for k, v in vars(self._args).items()
+            if k in supported_class_params and v is not None
+        }
+        print("\nclass params:")
+        pp.pprint(self._class_params)
+        print()
+
+    def _add_extra_arguments(self) -> None:
+        self._parser.add_argument(
+            "--no_cache",
+            action="store_true",
+            default=False,
+            help="whether to enable dataframe repartition, cache and cout outside fit function",
+        )
+
+    def score(
+        self,
+        transformed_df: DataFrame,
+        features_col: str,
+        prediction_col: str,
+    ) -> float:
+        """Computes the silhoutte score for the clustering result. This is a common metric to measure
+        how well the clustering algorithm performs.
+
+        Parameters
+        ----------
+        transformed_df
+            Model transformed data.
+        features_col
+            Name of features column.
+            Note: this column is assumed to be of pyspark sql 'array' type.
+        prediction_col
+            Name of prediction column
+
+        Returns
+        -------
+        float
+            The computed silhoutte score.
+
+        """
+        from sklearn.metrics import silhouette_score
+
+        sc = transformed_df.rdd.context
+
+        pdf: pd.DataFrame = transformed_df.toPandas()
+        features_pdf = pdf.drop(columns=[prediction_col])
+        prediction_pdf = pdf[prediction_col]
+
+        features_np = np.stack(features_pdf.to_numpy().squeeze())
+        prediction_np = prediction_pdf.to_numpy()
+
+        return silhouette_score(features_np, prediction_np)
```

Comment:
Curious how fast this runs for some examples.

Reply:
This is extremely slow for larger data size, so I typically comment this out when running for benchmark execution time of the algo

Reply:
Given that, maybe enable gpu version https://docs.rapids.ai/api/cuml/stable/api/ if gpu is detected, and otherwise if data is above a certain size, require an additional option to force computation?